spec/requests/api/user/financial_transactions_spec.rb

```diff
@@ -0,0 +1,109 @@
+require 'swagger_helper'
+
+describe 'User', type: :request do
+  include ApiHelper
+
+  let(:api_scopes) { ['finance:user'] }
+  let(:user) { create :user, groups: [create(:ordergroup)] }
+  let(:other_user2) { create :user }
+  let(:ft) { create(:financial_transaction, user: user, ordergroup: user.ordergroup) }
+
+  before do
+    ft
+  end
+
+  path '/user/financial_transactions' do
+    post 'create new financial transaction (requires enabled self service)' do
+      tags 'Financial Transaction'
+      consumes 'application/json'
+      produces 'application/json'
+
+      parameter name: :financial_transaction, in: :body, schema: {
+        type: :object,
+        properties: {
+          amount: { type: :integer },
+          financial_transaction_type: { type: :integer },
+          note: { type: :string }
+        }
+      }
+
+      let(:financial_transaction) { { amount: 3, financial_transaction_type_id: create(:financial_transaction_type).id, note: 'lirum larum' } }
+
+      response '200', 'success' do
+        schema type: :object, properties: {
+          financial_transaction: { '$ref': '#/components/schemas/FinancialTransaction' }
+        }
+        run_test!
+      end
+
+      it_handles_invalid_token_with_id
+      it_handles_invalid_scope_with_id 'user has no ordergroup, is below minimum balance, self service is disabled, or missing scope'
+
+      response '404', 'financial transaction type not found' do
+        schema '$ref' => '#/components/schemas/Error404'
+        let(:financial_transaction) { { amount: 3, financial_transaction_type_id: 'invalid', note: 'lirum larum' } }
+        run_test!
+      end
+
+      # TODO: fix controller to actually send a 422 for invalid params?
```

Comment:
My personal opinion about TODOs in the codebase is this: Never leave them in there unless there is at least an issue referenced to them. (E.g. `# TODO: https://github.com/foodcoops/foodsoft/issues/xxx - fix controller to actually send a 422 for invalid params?`) Otherwise these TODOs just stay in there forever in my experience.

Apart from that: Is commenting this test out not concealing a real bug?: _Shouldn't_ there be a validation error response when `note` is missing? (I truly don't know as I haven't got a lot of experience with this particular controller. :blush: )

Reply:
Yes, I think it's an actual bug, and we should fix it. But I also didn't want to throw it in this PR here ... I'll reference https://github.com/foodcoops/foodsoft/issues/999 there (nice issue number :D) and we can hopefully fix it soon ..